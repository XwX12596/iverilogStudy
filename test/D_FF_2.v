module d_ff(
    input wire clk,
    input wire D,
    input wire Rst,
    output reg Q
);

always @(negedge Rst or posedge clk)
begin
    if(!Rst)
        Q = 0;
    else
        Q= D;
end
endmodule

`timescale 1ns/1ps
module stimulus;
    reg clk;
    reg D;
    wire Q;
    reg Rst;
    d_ff d1(
        .clk(clk),
        .D(D),
        .Q(Q),
        .Rst(Rst)
    );
    integer i;
    initial begin
        $dumpfile("D_FF_2.vcd");
        $dumpvars(0,stimulus);
        D = 0;
        #8 D = 1;
        #10 D = 0;
        #10 D = 0;
        #10 D = 1;
        #10 D = 0;
        #10 D = 1;	
        #40;
    end
    initial begin
        clk = 0;
        for (i = 0; i<=10; i=i+1) begin
            #10 clk = ~clk;
        end
    end
    initial begin
        Rst = 1;
        #28 Rst = 0;
        #52 Rst = 1;
    end

endmodule