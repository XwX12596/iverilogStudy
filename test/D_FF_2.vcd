$date
	Tue Sep 20 16:33:04 2022
$end
$version
	Icarus Verilog
$end
$timescale
	1ps
$end
$scope module stimulus $end
$var wire 1 ! Q $end
$var reg 1 " D $end
$var reg 1 # Rst $end
$var reg 1 $ clk $end
$var integer 32 % i [31:0] $end
$scope module d1 $end
$var wire 1 " D $end
$var wire 1 # Rst $end
$var wire 1 $ clk $end
$var reg 1 ! Q $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
$end
#0
$dumpvars
b0 %
0$
1#
0"
x!
$end
#8000
1"
#10000
1!
b1 %
1$
#18000
0"
#20000
b10 %
0$
#28000
0!
0#
#30000
b11 %
1$
#38000
1"
#40000
b100 %
0$
#48000
0"
#50000
b101 %
1$
#58000
1"
#60000
b110 %
0$
#70000
b111 %
1$
#80000
b1000 %
0$
1#
#90000
1!
b1001 %
1$
#100000
b1010 %
0$
#110000
b1011 %
1$
