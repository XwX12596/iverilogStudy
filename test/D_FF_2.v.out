#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "D:\iverilog\lib\ivl\system.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "D:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "D:\iverilog\lib\ivl\va_math.vpi";
S_00000238f9086640 .scope module, "stimulus" "stimulus" 2 18;
 .timescale -9 -12;
v00000238f9086960_0 .var "D", 0 0;
v00000238f9086a00_0 .net "Q", 0 0, v00000238f8f96180_0;  1 drivers
v00000238f8fdc820_0 .var "Rst", 0 0;
v00000238f8fdc8c0_0 .var "clk", 0 0;
v00000238f8fdc960_0 .var/i "i", 31 0;
S_00000238f90867d0 .scope module, "d1" "d_ff" 2 23, 2 1 0, S_00000238f9086640;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 1 "D";
    .port_info 2 /INPUT 1 "Rst";
    .port_info 3 /OUTPUT 1 "Q";
v00000238f9086c80_0 .net "D", 0 0, v00000238f9086960_0;  1 drivers
v00000238f8f96180_0 .var "Q", 0 0;
v00000238f908bc60_0 .net "Rst", 0 0, v00000238f8fdc820_0;  1 drivers
v00000238f908bd00_0 .net "clk", 0 0, v00000238f8fdc8c0_0;  1 drivers
E_00000238f90890f0/0 .event negedge, v00000238f908bc60_0;
E_00000238f90890f0/1 .event posedge, v00000238f908bd00_0;
E_00000238f90890f0 .event/or E_00000238f90890f0/0, E_00000238f90890f0/1;
    .scope S_00000238f90867d0;
T_0 ;
    %wait E_00000238f90890f0;
    %load/vec4 v00000238f908bc60_0;
    %nor/r;
    %flag_set/vec4 8;
    %jmp/0xz  T_0.0, 8;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000238f8f96180_0, 0, 1;
    %jmp T_0.1;
T_0.0 ;
    %load/vec4 v00000238f9086c80_0;
    %store/vec4 v00000238f8f96180_0, 0, 1;
T_0.1 ;
    %jmp T_0;
    .thread T_0;
    .scope S_00000238f9086640;
T_1 ;
    %vpi_call 2 31 "$dumpfile", "D_FF_2.vcd" {0 0 0};
    %vpi_call 2 32 "$dumpvars", 32'sb00000000000000000000000000000000, S_00000238f9086640 {0 0 0};
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000238f9086960_0, 0, 1;
    %delay 8000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000238f9086960_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000238f9086960_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000238f9086960_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000238f9086960_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000238f9086960_0, 0, 1;
    %delay 10000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000238f9086960_0, 0, 1;
    %delay 40000, 0;
    %end;
    .thread T_1;
    .scope S_00000238f9086640;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000238f8fdc8c0_0, 0, 1;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v00000238f8fdc960_0, 0, 32;
T_2.0 ;
    %load/vec4 v00000238f8fdc960_0;
    %cmpi/s 10, 0, 32;
    %flag_or 5, 4;
    %jmp/0xz T_2.1, 5;
    %delay 10000, 0;
    %load/vec4 v00000238f8fdc8c0_0;
    %inv;
    %store/vec4 v00000238f8fdc8c0_0, 0, 1;
    %load/vec4 v00000238f8fdc960_0;
    %addi 1, 0, 32;
    %store/vec4 v00000238f8fdc960_0, 0, 32;
    %jmp T_2.0;
T_2.1 ;
    %end;
    .thread T_2;
    .scope S_00000238f9086640;
T_3 ;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000238f8fdc820_0, 0, 1;
    %delay 28000, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000238f8fdc820_0, 0, 1;
    %delay 52000, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v00000238f8fdc820_0, 0, 1;
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "D_FF_2.v";
